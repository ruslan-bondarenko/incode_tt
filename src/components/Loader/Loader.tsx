import { Flex, Spin, Typography } from "antd";
import React from "react";

const Loader = () => {
  return (
    <Flex gap="middle" align="center" justify="center">
      <Spin />
      <Typography.Text>Loading...</Typography.Text>
    </Flex>
  );
};

export default Loader;
