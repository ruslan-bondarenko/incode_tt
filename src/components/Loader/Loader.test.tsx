import { render, screen } from "@testing-library/react";
import React from "react";

import Loader from "./Loader";
import { describe, it } from "@jest/globals";

declare const expect: jest.Expect;

describe("Loader component", () => {
  it("Loader renders", () => {
    render(<Loader />);

    expect(screen.getByText("Loading...")).toBeInTheDocument();
  });

  it("Loader snapshot", () => {
    const view = render(<Loader />);

    expect(view).toMatchSnapshot();
  });
});
