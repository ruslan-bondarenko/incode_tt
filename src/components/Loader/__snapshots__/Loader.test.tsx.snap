// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Loader snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-justify-center ant-flex-gap-middle"
      >
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1kuana8"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
        <span
          class="ant-typography css-dev-only-do-not-override-1kuana8"
        >
          Loading...
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-justify-center ant-flex-gap-middle"
    >
      <div
        aria-busy="true"
        aria-live="polite"
        class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1kuana8"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
      <span
        class="ant-typography css-dev-only-do-not-override-1kuana8"
      >
        Loading...
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
